test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +91,10 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if(new Parser().parseCommand(inputCommand).isMutating()) {
```

Comment:
This change does not help as it does not confirm that saving happens only on mutating commands. Tests will pass even without this change. To be more complete, you should have an else branch to confirm that saving does not happen if the command is non mutating.